packages/typespec-client-generator-core/test/methods/lro.test.ts

```diff
@@ -0,0 +1,554 @@
+      // const pollingModel = runner.context.sdkPackage.models.find(
+      //   (m) => m.name === "ArmOperationStatusResourceProvisioningState"
+      // );
+      // ok(pollingModel);
+      // strictEqual(metadata.pollingStep.responseBody, pollingModel);
+      // TODO: TCGC bug to not include polling model https://github.com/Azure/typespec-azure/issues/1530
+      strictEqual(
+        metadata.pollingStep.responseBody?.name,
+        "ArmOperationStatusResourceProvisioningState"
+      );
+
+      strictEqual(metadata.finalResponse?.envelopeResult, roundtripModel);
+      strictEqual(metadata.finalResponse?.result, roundtripModel);
+      assert.isUndefined(metadata.finalResponse.resultPath);
+    });
+
+    it("ArmResourceDeleteWithoutOkAsync", async () => {
+      await runner.compileWithVersionedService(`
+        model Employee is TrackedResource<EmployeeProperties> {
+          ...ResourceNameParameter<Employee>;
+        }
+
+        model EmployeeProperties {
+          age?: int32;
+        }
+
+        #suppress "@azure-tools/typespec-azure-resource-manager/arm-post-operation-response-codes"
+        op delete is ArmResourceDeleteWithoutOkAsync<Employee>;
+    `);
+      const roundtripModel = runner.context.sdkPackage.models.find((m) => m.name === "Employee");
+      assert.isUndefined(roundtripModel);
+      const methods = runner.context.sdkPackage.clients[0].methods;
+      strictEqual(methods.length, 1);
+      const method = methods[0];
+      strictEqual(method.name, "delete");
+      assert.include(
+        method.parameters.map((m) => m.name),
+        "employeeName"
+      );
+      assert.notInclude(
+        method.parameters.map((m) => m.name),
+        "resource"
+      );
+
+      const metadata = (method as SdkLroServiceMethod<SdkHttpOperation>).lroMetadata;
+      ok(metadata);
+      strictEqual(metadata.finalStateVia, FinalStateValue.location);
+      strictEqual(metadata.finalStep?.kind, "finalOperationLink");
+
+      // ARM LRO core types are different
+      // const pollingModel = runner.context.sdkPackage.models.find(
+      //   (m) => m.name === "ArmOperationStatusResourceProvisioningState"
+      // );
+      // ok(pollingModel);
+      // strictEqual(metadata.pollingStep.responseBody, pollingModel);
+      // TODO: TCGC bug to not include polling model
+      strictEqual(
+        metadata.pollingStep.responseBody?.name,
+        "ArmOperationStatusResourceProvisioningState"
+      );
+
+      assert.isUndefined(metadata.finalResponse);
+    });
+
+    it("ArmResourceActionAsync", async () => {
+      await runner.compileWithVersionedService(`
+        model Employee is TrackedResource<EmployeeProperties> {
+          ...ResourceNameParameter<Employee>;
+        }
+
+        model EmployeeProperties {
+          /** Age of employee */
+          age?: int32;
+        }
+
+        #suppress "@azure-tools/typespec-azure-resource-manager/arm-post-operation-response-codes"
```

Comment:
nit, does this suppress required when using ARM template?

Reply:
I actually copied from here: https://github.com/Azure/typespec-azure/blob/dfeee05c8fe3ec61ff28daeb0db59942ed08f8fe/packages/samples/specs/resource-manager/operations/nobody-action/main.tsp#L85-L86

Anyway, I deleted all `#suppress` and the test cases are OK. Not sure if all those `#suppress` are unnecessary or the test framework has suppressed all warnings by default. @XiaofeiCao FYI